packages/contracts/contracts/verifierHelpers/BaseVerifierHelper.sol

```diff
@@ -8,64 +8,93 @@ contract BaseVerifierHelper {
 
     uint256 public constant SNARK_SCALAR_FIELD =
         21888242871839275222246405745257275088548364400416034343698204186575808495617;
+    uint256 public immutable chainid;
 
     struct EpochKeySignals {
-        bool revealNonce;
```

Comment:
Try to keep the smaller types near each other in the `struct` definition. If the types can fit into a single 32 byte area the compiler will do so automatically and use less memory.

e.g.

```
bool a;
uint48 b;
uint160 c;
uint256 d;
```

instead of 

```
bool a;
uint48 b;
uint256 d;
uint160 c;
```

Reply:
I tried several combinations
the current one is the best ( but not sure what it is the optimized)
| | before |  [bc03229](https://github.com/Unirep/Unirep/pull/547/commits/bc03229685101249640c3442bb703cc989e971ec)|
| :--:| :--:| :--:|
| Unirep.sol | 3612695 | 3599327 |
| EpochKeyVerifierHelper.sol | 616423 | 615775 |
| EpochKeyLiteVerifierHelper.sol | 608769 | 607695 |
| ReputationVerifierHelper.sol | 780606 | 772772 |